phoenix-shared/src/commonMain/kotlin/fr.acinq.phoenix/db/payments/MetadataQueries.kt

```diff
@@ -0,0 +1,120 @@
+        data: WalletPaymentMetadataRow
+    ) {
+        queries.addMetadata(
+            type = id.dbType.value,
+            id = id.dbId,
+            lnurl_base_type = data.lnurl_base?.first,
+            lnurl_base_blob = data.lnurl_base?.second,
+            lnurl_description = data.lnurl_description,
+            lnurl_metadata_type = data.lnurl_metadata?.first,
+            lnurl_metadata_blob = data.lnurl_metadata?.second,
+            lnurl_successAction_type = data.lnurl_successAction?.first,
+            lnurl_successAction_blob = data.lnurl_successAction?.second,
+            user_description = data.user_description
+        )
+    }
+
+    fun getMetadata(
+        id: WalletPaymentId,
+        options: WalletPaymentFetchOptions
+    ): WalletPaymentMetadata? {
+        // Optimization notes:
+        // We have a reason to optimize fetching descriptions, because it's commonly used.
+        // However, at this point, other custom options are uncommon.
+        // So we're not optimizing other combinations at this point,
+        // but the possibility is there for future use.
+        return when (options) {
+            WalletPaymentFetchOptions.None -> {
+                null
+            }
+            WalletPaymentFetchOptions.Descriptions -> {
+                getMetadataDescriptions(id)
+            }
+            else -> {
+                getMetadataAll(id)
+            }
+        }
+    }
+
+    private fun getMetadataDescriptions(id: WalletPaymentId): WalletPaymentMetadata? {
+        return queries.fetchDescriptions(
+            type = id.dbType.value,
+            id = id.dbId,
+            mapper = ::mapDescriptions
+        ).executeAsOneOrNull()
+    }
+
+    private fun getMetadataAll(id: WalletPaymentId): WalletPaymentMetadata? {
+        return queries.fetchMetadata(
+            type = id.dbType.value,
+            id = id.dbId,
+            mapper = ::mapAll
+        ).executeAsOneOrNull()
+    }
+
+    companion object {
+        fun mapDescriptions(
+            lnurl_description: String?,
+            user_description: String?
+        ): WalletPaymentMetadata {
+            return WalletPaymentMetadata(
+                lnurlDescription = lnurl_description,
+                userDescription = user_description
+            )
+        }
+
+        @Suppress("UNUSED_PARAMETER")
+        fun mapAll(
+            type: Long,
+            id: String,
+            lnurl_base_type: LNUrlBase.TypeVersion?,
+            lnurl_base_blob: ByteArray?,
+            lnurl_description: String?,
+            lnurl_metadata_type: LNUrlMetadata.TypeVersion?,
+            lnurl_metadata_blob: ByteArray?,
+            lnurl_successAction_type: LNUrlSuccessAction.TypeVersion?,
+            lnurl_successAction_blob: ByteArray?,
+            user_description: String?
+        ): WalletPaymentMetadata {
+            val lnurl_base = if(
+                lnurl_base_type != null && lnurl_base_blob != null
+            ) {
```

Comment:
Indentation looks a bit strange, why not let the if be on a single line ? 
```kotlin
if (lnurl_base_type != null && lnurl_base_blob != null) {
  lnurl_base_type to lnurl_base_blob
} else null
```

^ looks a bit better, wdyt?

Reply:
Yup, looks better. Done in 2675f7f